### PR TITLE
[fluentd-elasticsearch] Allow elasticsearch type_name to be specified, optionally

### PR DESCRIPTION
**Pull request by 0x4c6565**
Signed-off-by: Lee Spottiswood <lee.spottiswood@ukfast.co.uk>

#### What this PR does / why we need it:
This PR allows the `type_name` directive to be specified, and omitted when value is empty

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://developercertificate.org) signed
- [x] Chart Version bumped (if the pr is an update to an existing chart)
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[fluentd-elasticsearch]`)

### CODE REVIEW

File: charts/fluentd-elasticsearch/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: fluentd-elasticsearch
-version: 4.3.2
+version: 4.3.3
```
Comment (monotek) [inline charts/fluentd-elasticsearch/Chart.yaml]: new feature added. minor version should be raised (4.4.0)

File: charts/fluentd-elasticsearch/templates/configmaps.yaml [hunk 2]
```diff
@@ -472,12 +472,14 @@ data:
       @type elasticsearch
       @log_level info
       include_tag_key true
-      type_name _doc
       host "#{ENV['OUTPUT_HOST']}"
       port "#{ENV['OUTPUT_PORT']}"
       scheme "#{ENV['OUTPUT_SCHEME']}"
       ssl_verify "#{ENV['OUTPUT_SSL_VERIFY']}"
       ssl_version "#{ENV['OUTPUT_SSL_VERSION']}"
+{{- if (ne .Values.elasticsearch.typeName "") }}
```
Comment (monotek) [inline charts/fluentd-elasticsearch/templates/configmaps.yaml]: not sure if this makes sense as a default is set in the values.yaml and therfore its always true...

**Comment by monotek** [comment]
Please also check how this will affect updates as the value is new for users using an older chart version.

File: charts/fluentd-elasticsearch/templates/configmaps.yaml (hunk 2 above)
Comment (0x4c6565) [reply charts/fluentd-elasticsearch/templates/configmaps.yaml]: In my use case, I have overridden the value with "". The logic above allows this to be omitted when the value is overridden with an empty string.

In this situation, the elasticsearch plugin defaults to type `fluentd`

**Comment by 0x4c6565** [comment]
> 
> 
> Please also check how this will affect updates as the value is new for users using an older chart version.

The default for this value is `_doc`, which is the value of the current hardcoded `type_name`.

**Pull request merged**